posts/2020/03/pytorch-tensoriterator-internals.md

```diff
@@ -45,16 +47,13 @@ an essential part of learning to contribute to the pytorch codebase since iterat
 over tensors in the C++ codebase are extremely commonplace. This post is aimed at someone
 who wants to contribute to pytorch, and you should at least be familiar with some of the
 basic terminologies of the pytorch codebase that can be found in Edward Yang's 
-execellent [blog post](http://blog.ezyang.com/2019/05/pytorch-internals/**) on pytorch internals.
+excellent [blog post](http://blog.ezyang.com/2019/05/pytorch-internals/**) on pytorch internals.
 
 # History of TensorIterator
 
 ## TH iterators
 
-`TensorIterator` was incorporated into the `ATen` implementation of pytorch tensors when the
-pytorch team decided to change the C macro-based `TH` implementation and use a templated
-C++ implementation instead. Previously in `TH`, C macros would be used for writing tensor
-loops in a type independent manner. For example, consider this simple `TH` loop
+TensorIterator was devised to simplify the implementation of pytorch's tensor operations over the TH implementation. TH uses preprocessor macros to write type-independent loops over tensors, instead of C++ templates. For example, consider this simple `TH` loop
```

Comment:
```suggestion
TensorIterator was devised to simplify the implementation of pytorch's tensor operations over the `TH` implementation. `TH` uses preprocessor macros to write type-independent loops over tensors, instead of C++ templates. For example, consider this simple `TH` loop
```